fgpyo/util/types.py

```diff
@@ -73,9 +84,52 @@ def is_constructible_from_str(type_: type) -> bool:
     return False
 
 
-def _is_optional(type_: type) -> bool:
-    """Returns true if type_ is optional"""
-    return typing.get_origin(type_) is Union and type(None) in typing.get_args(type_)
+# NB: since `_GenericAlias` is a private attribute of the `typing` module, mypy doesn't find it
+TypeAnnotation: TypeAlias = Union[type, typing._GenericAlias, UnionType]  # type: ignore[name-defined]
```

Comment:
**issue** `GenericAlias` should be included as a valid type annotation type. Also add `from types import GenericAlias` above

```suggestion
TypeAnnotation: TypeAlias = Union[type, typing._GenericAlias, UnionType, 
types.GenericAlias]  # type: ignore[name-defined]
```